src/h3/memview_int.pyx

```diff
@@ -147,6 +148,10 @@ cpdef H3int[:] children(H3int h, res=None):
 
 
 cpdef H3int[:] compact(const H3int[:] hu):
+    # todo: gotta be a more elegant way to handle these...
+    if len(hu) == 0:
+        return empty_memory_view()
```

Comment:
What goes wrong if `hu` is zero length? If it's a core library crash, that should be addressed by the yet unreleased uber/h3#278

Reply:
The bug appears when I run the test suite (without the above `if` block) as:

```python
E   IndexError: Out of bounds on buffer access (axis 0)
memview_int.pyx:207: IndexError
```

It looks like it comes down to this expression: `&hc[0]`. This is the standard syntax for Cython, so that's tricky to avoid.

It sounds like the change in uber/h3#278 was to not dereference the input pointer if the length was zero. Is that right?

If that's the case, given that the Cython syntax essentially **requires** dereferencing, I think the current solution which short-circuits the whole mess is probably the easiest.

Anyone see a cleaner solution?

@isaacbrodsky @nrabinowitz @dfellis @karentycoon

Reply:
Maybe add a brief comment explaining this is good. This solution seems good.